### ISSUE TITLE
[Bug] Addon stops working and responding on Chrome after a while

### ISSUE THREAD

**Issue by rpgdev** [darkreader/darkreader#8368]
# Bug Report

## Website Address

N/A

## Bug Description
After a while (or opening several tabs), darkreader just stops working and responding on Chrome. No websites are automatically "darked" and clicking the addon button doesn't result in any response from it--no popup menu shows up. It has been happening for a while (months if not more than a year) but I decided to report it now. 

## Steps To Reproduce

- Go on reddit.
- Open several tabs normally. 
- Eventually the addon just stops working/responding

## Expected Behavior

Addon to keep working.

## Actual Behavior

Addon stops working after a while (or several tabs opened, not sure what triggers it)

## Screenshots

N/A

## System Information

- OS: Fedora 34 KDE Plasma
- Browser: Chrome Version 98.0.4758.80 (Official Build) (64-bit)
- Dark Reader Version: 4.9.45

## Additional Context

N/A

**Comment by Gusted** [comment]
Well these more sounds like a browser bug, but just to be sure.

A few things:
1. Are you running low on memory when dark reader doesn't respond anymore?
2. Do you have any other extensions, and do they still work when Dark Reader doesn't respond anymore?
3. Are there any reported errors in the background script? `chrome://extensions` -> Enable Developer Mode -> Details(under Dark Reader extension "card") -> Inspect background -> Console

**Comment by rpgdev** [comment]
1. I don't suffer from having low memory (have around 40gb).
2. Other extensions keep working, just checked since right now darkreader is "crashed".
3. I don't see "inspect background" I see "inspect views" which points to "background/index.html". I click that and go to the console but there's no output. I refresh a webpage and don't see anything there, am I supposed to see something? Am I in the right place?

**Comment by Gusted** [comment]
At the very least you should be seeing:
![image](https://user-images.githubusercontent.com/25481501/156921831-9dd9e352-2b25-42aa-9afb-a7581b1003a9.png)

**Comment by rpgdev** [comment]
There's nada. Maybe it would show up if I restart that browser instance since the addon would start working again. But would disappear once it crashes again. Let me know what would be the best way to help debug it.

**Comment by Gusted** [comment]
If that doesn't show up then it's most likely a browser bug of some sort, as that's a unconditionally `console.log` and there's no way to remove such message unless the extension is restarted of some sort.

**Comment by patheticGeek** [comment]
Can confirm this issue, started experiencing this on brave a few days ago.
The extension just doesn't respond to any button clicks or set the right theme on pages.

Currently seems like after re-installing it is working fine, will update my comment if this comes back.

**Comment by Gusted** [comment]
I'm considering open a bug report on chromium bug tracker for this, as mainly because the logged message:

>  ![image](https://user-images.githubusercontent.com/25481501/156921831-9dd9e352-2b25-42aa-9afb-a7581b1003a9.png)

Isn't shown anymore after it "crashes" which is quite weird and can _to the best of my knowledge_ only happen due to some browser behavior.

**Comment by rpgdev** [comment]
@Gusted were you able to reproduce it on your end?

**Comment by Gusted** [comment]
I don't daily drive chromium, but given the symptoms and a friend of my also complaining that it happens out of the blue just recently doesn't give me a need to also try to reproduce it, Dark Reader was last updated over a month ago, really strange if a bug from dark reader just now got reported, that would be a weird coincidence.

**Comment by rpgdev** [comment]
Weird. Like I said in the bug report I've been experiencing this for more than a year definitely. I actually assumed someone would point me to a previously submitted bug report of the same issue right after I created this one.

**Comment by Gusted** [comment]
> I actually assumed someone would point me to a previously submitted bug report of the same issue right after I created this one.

Well first time I've heard of this, and the fact that it has been happening for longer doesn't really help narrowing it down to anything for what's worth it could be a very very old regression of dark reader.

**Comment by cjauvin** [comment]
I use Dark Reader with Chrome and Firefox on Ubuntu 20.04 (a relatively powerful gaming desktop with plenty of ressources), and I've had this exact issue, with Chrome specifically, for a long while (Firefox never had any problem). It almost became an habit for me, every 10 tabs or so I open, to right-click on the extension icon, choose "manage extension", and trigger the enable toggle off and on again, which fixes it, until the next "crash" (which is really more a "lock up"). One thing that never became clear however are the particular sites or patterns triggering the crash. I suspect that it's related to Google apps (I always have a few of them pinned), but it's not systematic enough to be certain. I can try to use the context of this issue to make an extra effort and try to provide helpful information.

**Comment by rpgdev** [comment]
@cjauvin Can you please clarify what you mean by "google apps"? Just the sites opened or PWAs? In my case I usually don't have any google app or site opened since I use that chrome profile to do some leisure reddit browsing. Some tabs do have youtube (with dark theme) opened often. From now on I will pay closer attention to see if youtube triggers it.

**Comment by cjauvin** [comment]
My Chrome opens with 4 pinned tabs: Gmail, GCal, Outlook mail and Outlook calendar. Regularly, throughout the day, I also open a "chat.google.com" tab, not pinned.

**Comment by Gusted** [comment]
I'm not sure how Chrome handles logging things, but when such "crash" happens it might log something(and might indicate the reason why Dark reader crashes) as from my experience and looking at this report it seems to indicate that dark reader is somehow making Chrome unhappy in which they respond with "crashing"(or simply disabling dark reader?)

**Comment by cjauvin** [comment]
I activated error logging and I just got that:

![image](https://user-images.githubusercontent.com/488992/157889945-bd3b93a3-497b-4f0c-ab0b-3739db53313d.png)

**Comment by cjauvin** [comment]
Got this today : 

![image](https://user-images.githubusercontent.com/488992/158841441-b6bda8a7-986a-491e-914c-cfb3e8633f1c.png)

**Comment by rpgdev** [comment]
@Gusted when will the update trickle down to the Chrome addon store? :)

**Comment by Gusted** [comment]
Releases are usually around the beginning of the month

**Comment by eldorel** [comment]
> > I actually assumed someone would point me to a previously submitted bug report of the same issue right after I created this one.
> 
> Well first time I've heard of this, and the fact that it has been happening for longer doesn't really help narrowing it down to anything for what's worth it could be a very very old regression of dark reader.

FYI I reported the same behavior [here](https://github.com/darkreader/darkreader/issues/7569#issuecomment-989599493) thinking it was related to a different bug

**Comment by cjauvin** [comment]
With version 4.9.47 on Chrome (Linux), I'm still experimenting the problem described in this thread, but now I'm not seeing any error in the logs anymore.

**Comment by eldorel** [comment]
4.9.47 was released prior to the commit that allegedly addresses this issue.

I'm still seeing the problem as well.
I've been patiently waiting for 4.9.58 to release for a couple of weeks,
but I've started looking for other solutions.

Having Darkreader fail randomly is literally playing russian roulette with
my migraines, and there are some work related tasks that I *can't* do
without chrome.

This *is* a major bug for me, but the dev just doesn't seem to actually
care. (or they don't appreciate the frequency of the failures)

Either way, i'm shopping for a physical screen filter and a new software
solution in case the next release doesn't fix it or continues to be delayed.
